src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
-import React from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { setUserFromApiById } from '../../utils/setUserFromApiById';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal = () => {
+  const [user, setUser] = useState<User>();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const selectedTodoDispatch = useAppDispatch();
+
+  const changeCurrentTodo = useCallback(
+    (newTodos: Todo | null) => {
+      selectedTodoDispatch(
+        currentTodoSlice.actions.changecurrentTodo(newTodos),
```

Comment:
There is a typo in the action name `changecurrentTodo`. It should be `changeCurrentTodo` to match the typical camelCase convention and ensure consistency with the action defined in the `currentTodoSlice`.